### PR TITLE
Responsive line chart

### PR DESCRIPTION
**Pull request by nulikartik**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes
1)Making the line chart responsive
2) Correcting flex behavior
3)Smoothing ends of lines to make the intersection between lines smooth
Single line chart with numeric axis:
image
Single line chart with string axis:
image
Multiple line chart:
image
Multiple line chart with isolation state:
image

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/charting/src/components/LineChart/LineChart.base.tsx [hunk 1]
```diff
@@ -1,91 +1,81 @@
...
+  private xAxisElement: SVGElement | null;
+  private yAxisElement: SVGElement | null;
+  // tslint:disable-next-line:no-any
+  private _xAxisScale: any = '';
+  // tslint:disable-next-line:no-any
+  private _yAxisScale: any = '';
   private _uniqLineText: string;
-
+  private chartContainer: HTMLDivElement;
+  private margins = { top: 20, right: 10, bottom: 35, left: 40 };
```
Comment (Raghurk) [inline packages/charting/src/components/LineChart/LineChart.base.tsx]: >private margins = { top: 20, right: 10, bottom: 35, left: 40 }; [](start = 2, length = 63)

comment as to why you need this.

Comment (nulikartik) [reply packages/charting/src/components/LineChart/LineChart.base.tsx]: added comments

File: packages/charting/src/components/LineChart/LineChart.styles.ts [hunk 2]
```diff
@@ -1,116 +1,74 @@
...
     ],
     calloutContentY: [
       {
-        fontSize: fonts.xxLarge.fontSize,
-        fontFamily: fonts.xxLarge.fontFamily,
-        lineHeight: '33px'
+        color: color === '' ? palette.black : color,
+        fontSize: '28px',
+        fontFamily: 'Segoe UI',
+        fontWeight: 'bold',
```
Comment (kkjeer) [inline packages/charting/src/components/LineChart/LineChart.styles.ts]: Use `fonts` styles here

File: packages/charting/src/components/LineChart/LineChart.styles.ts [hunk 3]
```diff
@@ -1,116 +1,74 @@
...
-    ],
-    chartLabel: [
-      {
-        textAlign: 'center',
-        ...theme.fonts.mediumPlus
+    xAxis: {
+      selectors: {
+        text: {
+          opacity: 0.6,
+          fontSize: '10px'
```
Comment (kkjeer) [inline packages/charting/src/components/LineChart/LineChart.styles.ts]: Use `fonts` sizing

File: packages/charting/src/components/LineChart/LineChart.styles.ts (hunk 2 above)
Comment (nulikartik) [reply packages/charting/src/components/LineChart/LineChart.styles.ts]: Updated.

**Pull request merged**